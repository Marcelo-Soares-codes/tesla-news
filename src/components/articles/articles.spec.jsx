// Articles.test.js

import React from 'react';
import { render, screen } from '@testing-library/react';
import { Articles } from './index'; // Ajuste o caminho conforme a estrutura do seu projeto.

// Mock do componente Card
jest.mock('./cards', () => ({
  Card: ({ title }) => <div data-testid={`mock-card-${title.replace(/\s+/g, '-').toLowerCase()}`}>{title}</div>,
}));

describe('Articles Component', () => {
  it('renders Articles correctly with articles', () => {
    const mockArticles = [
      {
        url: 'https://example.com/1',
        urlToImage: 'https://example.com/image1.jpg',
        title: 'Article 1',
        description: 'Description 1',
        author: 'Author 1',
      },
      {
        url: 'https://example.com/2',
        urlToImage: 'https://example.com/image2.jpg',
        title: 'Article 2',
        description: 'Description 2',
        author: 'Author 2',
      },
    ];

    render(<Articles articles={mockArticles} />);

    // Verifique se o Card component é chamado para cada artigo
    mockArticles.forEach(article => {
      expect(screen.getByTestId(`mock-card-${article.title.replace(/\s+/g, '-').toLowerCase()}`)).toBeInTheDocument();
    });

    // Salve um snapshot do componente renderizado
    expect(screen.getByTestId('mock-card-article-1')).toMatchSnapshot();
  });

  it('renders Articles correctly with removed article', () => {
    const mockArticles = [
      {
        url: 'https://example.com/1',
        urlToImage: 'https://example.com/image1.jpg',
        title: '[Removed]',
        description: 'Description 1',
        author: 'Author 1',
      },
      {
        url: 'https://example.com/2',
        urlToImage: 'https://example.com/image2.jpg',
        title: 'Article 2',
        description: 'Description 2',
        author: 'Author 2',
      },
    ];
  
    render(<Articles articles={mockArticles} />);
  
    // Verifique se pelo menos um Card component foi renderizado
    const cards = screen.queryAllByTestId(/^mock-card-.+/);
  
    // Filtra os cards que foram renderizados
    const renderedCards = cards.filter(card => card.textContent.includes('Article'));
  
    // Continue com a verificação do comprimento
    expect(renderedCards).toHaveLength(mockArticles.length - 1);
  
    // Salve um snapshot do componente renderizado
    // Aqui, você pode usar o snapshot do primeiro card, pois é o único renderizado
    const firstCard = renderedCards[0];
    expect(firstCard).toMatchSnapshot();
  });
});
