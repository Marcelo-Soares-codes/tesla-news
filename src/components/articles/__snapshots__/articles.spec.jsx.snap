// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles Component renders Articles correctly with articles 1`] = `
<div
  data-testid="mock-card-article-1"
>
  Article 1
</div>
`;

exports[`Articles Component renders Articles correctly with removed article 1`] = `
<div
  data-testid="mock-card-article-2"
>
  Article 2
</div>
`;
